src/panel_material_ui/widgets/SelectPicker.jsx

```diff
@@ -0,0 +1,185 @@
+import {useState} from 'react';
+import InputAdornment from '@mui/material/InputAdornment';
+import FilterListIcon from '@mui/icons-material/FilterList';
+import ClearIcon from '@mui/icons-material/Clear';
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+import MenuItem from '@mui/material/MenuItem';
+import Checkbox from '@mui/material/Checkbox';
+import ListItemText from '@mui/material/ListItemText';
+import FormControl from '@mui/material/FormControl';
+import InputLabel from '@mui/material/InputLabel';
+import Select from '@mui/material/Select';
+import OutlinedInput from '@mui/material/OutlinedInput';
+
+export function render({ model, el }) {
+    
+    const [title] = model.useState("title");
+    const [label] = model.useState("label");
+    const [options] = model.useState("options");
+    const [optionsLabels] = model.useState("options_labels");
+    const [value, setValue] = model.useState("value");
+    const [filterStr, setFilterStr] = model.useState("filter_str");
+    
+    const [width] = model.useState("width");
+    const [dropdownHeight] = model.useState("dropdown_height");
+    
+    // Store anchorPosition in state to trigger re-renders
+    const [anchorPosition, setAnchorPosition] = useState({ top: 500, left: 50 });
+
+    const handleClick = (event) => {
+        const elRect = el.getBoundingClientRect();
+        const parentRect = document.getElementsByTagName("fast-card")[0]?.getBoundingClientRect() || { top: 0, left: 0 };
```

Comment:
What's this about?

Reply:
It's a tweak I had to write, to make the dropdown list show up at the right coordinates using the FastListTemplate. Probably not required here. I had removed a few things related to FastListTemplate but have definitely missed this one.

Reply:
I get it, but will have to update with a proper fix.

Reply:
Yes please, I'd like to know a better way.

Reply:
Mine is not much less hacky.

Reply:
Is the goal to make the widget work with any existing Panel template ?